apps/web/src/app/share/[slug]/page.tsx

```diff
@@ -0,0 +1,20 @@
+import { notFound, permanentRedirect } from 'next/navigation';
+import { getLongURL } from '../_actions/get-long-url';
+import { updateClick } from '../_actions/update-click-count';
+
+type Props = {
+  params: {
+    slug: string;
+  };
+};
+
+export default async function Page({ params }: Props) {
+  const { slug } = params;
+  const longURL = await getLongURL(slug);
```

Comment:
what’s this page for

Reply:
Redirect to the long url. We need to catch the url somehow and decide + redirect it, we can either do like this or I think maybe in the middleware.

Reply:
ohh right! do we need to add anything to metadata or robots to make sure this can't be crawled. or is that not a big deal

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308 .I'm doing a permanent redirect, the robots will update their records that the url is moved, so there will be no issue most prob.